### PR TITLE
add handy method for get json with credential

### PR DESCRIPTION
**Pull request by lk9100**

### CODE REVIEW

File: lib/serverpipe.js [hunk 1]
```diff
@@ -36,6 +36,23 @@ export function fetchWithCredentials(url, options = {}) {
...
 }
 
+export function getJsonWithCredentials(url, body, options = {}) {
+    const defaultHeaders = {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+    };
+    const headers = 'headers' in options ? [...options.headers, defaultHeaders] : defaultHeaders;
+
+    headers[getCsrfHeaderName()] = getCsrfToken();
```
Comment (lol768) [inline lib/serverpipe.js]: For a GET?

**Review by lol768 — CHANGES_REQUESTED** [review]

File: lib/serverpipe.js (hunk 1 above)
Comment (lk9100) [reply lib/serverpipe.js]: yes

File: lib/serverpipe.js [hunk 2]
```diff
@@ -36,6 +36,23 @@ export function fetchWithCredentials(url, options = {}) {
...
+    return fetchWithCredentials(url, {
+        method: 'GET',
+        body: JSON.stringify(body),
```
Comment (lol768) [inline lib/serverpipe.js]: GET body? :thinking:

**Review by lol768 — CHANGES_REQUESTED** [review]

File: lib/serverpipe.js (hunk 1 above)
Comment (lol768) [reply lib/serverpipe.js]: The CSRF filter shouldn't execute for GETs?

File: lib/serverpipe.js (hunk 2 above)
Comment (lk9100) [reply lib/serverpipe.js]: hmm... why did i do this

Comment (lk9100) [reply lib/serverpipe.js]: killed that body

File: lib/serverpipe.js (hunk 1 above)
Comment (lk9100) [reply lib/serverpipe.js]: should we? maybe other server end could be checking csrf header? (this is extracted from other project btw..)

Comment (lol768) [reply lib/serverpipe.js]: A GET endpoint shouldn't ever change state, so there should no risk of a cross-site request causing damage.

Comment (lk9100) [reply lib/serverpipe.js]: 🌊

File: lib/serverpipe.js [hunk 3]
```diff
@@ -36,6 +36,21 @@ export function fetchWithCredentials(url, options = {}) {
     );
 }
 
+export function getJsonWithCredentials(url, options = {}) {
+    const defaultHeaders = {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
```
Comment (lol768) [inline lib/serverpipe.js]: Why?

File: lib/serverpipe.js [hunk 4]
```diff
@@ -36,6 +36,21 @@ export function fetchWithCredentials(url, options = {}) {
     );
 }
 
+export function getJsonWithCredentials(url, options = {}) {
+    const defaultHeaders = {
+        Accept: 'application/json',
```
Comment (lol768) [inline lib/serverpipe.js:41]:
```suggestion
        'Accept': 'application/json',
```

**Review by lol768 — CHANGES_REQUESTED** [review]

File: lib/serverpipe.js (hunk 3 above)
Comment (lk9100) [reply lib/serverpipe.js]: 💀

File: lib/serverpipe.js [hunk 5]
```diff
@@ -36,12 +36,26 @@ export function fetchWithCredentials(url, options = {}) {
     );
 }
 
-export function postJsonWithCredentials(url, body, options = {}) {
+export function getJsonWithCredentials(url, options = {}) {
     const defaultHeaders = {
         Accept: 'application/json',
```
Comment (lol768) [inline lib/serverpipe.js:41]:
```suggestion
        'Accept': 'application/json',
```

**Review by lol768 — CHANGES_REQUESTED** [review]

**Review by cursley — APPROVED** [review]

**Review by lol768 — APPROVED** [review]

**Pull request merged**